chunk/chunk_store.go

```diff
@@ -980,9 +924,12 @@ func (r *dynamoBatchWriteItemsOp) do() {
 		}
 
 		// If there are unprocessed items, backoff and retry those items.
-		if resp.UnprocessedItems != nil && len(resp.UnprocessedItems[r.tableName]) > 0 {
-			dynamoUnprocessedItems.Add(float64(len(resp.UnprocessedItems[r.tableName])))
-			unprocessed = append(unprocessed, resp.UnprocessedItems[r.tableName]...)
+		if resp.UnprocessedItems != nil && dictLen(resp.UnprocessedItems) > 0 {
```

Comment:
The reality is most of the time we'll only be writing to a single table, and in the worse case we'll be writing to two (when we are on the edge of two tables).  This is nice a clear so I think I'd prefer to leave it as is.